factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/validation/http/HttpSchemaRegistry.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.store.pgsql.validation.http;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.factcast.store.pgsql.validation.schema.SchemaKey;
+import org.factcast.store.pgsql.validation.schema.SchemaRegistry;
+import org.factcast.store.pgsql.validation.schema.SchemaRegistryUnavailableException;
+import org.factcast.store.pgsql.validation.schema.SchemaSource;
+import org.factcast.store.pgsql.validation.schema.SchemaStore;
+import org.springframework.stereotype.Component;
+
+import com.github.fge.jsonschema.core.exceptions.ProcessingException;
+import com.github.fge.jsonschema.main.JsonSchema;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Stopwatch;
+import com.google.common.cache.AbstractLoadingCache;
+import com.google.common.cache.LoadingCache;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Component
+
+public class HttpSchemaRegistry implements SchemaRegistry {
+
+    private IndexFetcher indexFetcher;
+
+    private SchemaFetcher schemaFetcher;
+
+    public HttpSchemaRegistry(URL baseUrl, SchemaStore store) {
+        this.store = store;
+        this.indexFetcher = new IndexFetcher(baseUrl);
+        this.schemaFetcher = new SchemaFetcher(baseUrl);
+    }
+
+    @VisibleForTesting
+    protected HttpSchemaRegistry(SchemaStore store, IndexFetcher indexFetcher,
+            SchemaFetcher schemaFetcher) {
+        this.store = store;
+        this.indexFetcher = indexFetcher;
+        this.schemaFetcher = schemaFetcher;
+    }
+
+    @NonNull
+    private final SchemaStore store;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=4877539)